tests/db_functions/test_cast.py

```diff
@@ -42,6 +44,15 @@ def test_cast_to_integer(self):
                 numbers = Author.objects.annotate(cast_int=Cast('alias', field_class()))
                 self.assertEqual(numbers.get().cast_int, 1)
 
+    def test_cast_from_python_to_date(self):
+        for field_class, value in (
+            (models.DateField, datetime.date.today()),
+            (models.DateTimeField, datetime.datetime.now().replace(microsecond=0)),
```

Comment:
Why `replace(microsecond=0)` is needed?

Reply:
Microseconds disappear on Oracle, it's a separate issue. I will try to investigate and solve it at the weekend.

Reply:
@sir-sigurd I added explicit workaround on Oracle that will be removed after [#28934](https://code.djangoproject.com/ticket/28934) fix. IMO this PR should be merged in advance.